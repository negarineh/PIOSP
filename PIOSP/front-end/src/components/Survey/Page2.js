import React from 'react';
import { PropTypes } from 'prop-types';
import '../stylesheets/Page2.css';
import '../stylesheets/MainPage.css'; 
import '../stylesheets/fonts/font-awesome-4.7.0/scss/font-awesome.css';
import '../stylesheets/fonts/font-awesome-4.7.0/less/animated.less';
// import Modal1 from 'react-modal';
// import Loading from 'react-loading-components';
// import { confirmAlert } from 'react-confirm-alert'; // Import
import '../stylesheets/react-confirm-alert.css' // Import css

import { Segment, Icon, Step, Modal} from 'semantic-ui-react';

// second page of survey
// In this page we are receiving name and email of participant and sending through server
// after this page we have participant pending page 
class Page2 extends React.Component {
    constructor(props) {
      super(props);   
      this.state={
        username: '',
        email: '',
        // modalIsOpen: false,
      };
      this.handleSubmit = this.handleSubmit.bind(this);

    }

    handleSubmit(e) {
      e.preventDefault();
      const data = new FormData(e.target);
      
      const curUsername = data.get('username'),
            curEmail = data.get('email');

      // On submit of the form, send a POST request with the data to the server.
    fetch(`signupparticipant`,{
      method: 'POST',
      body:JSON.stringify({
        username: curUsername,
        email: curEmail,
        // modalIsOpen: true
      }),
      headers: {"Content-Type": "application/json"}
    })
    .then(function(response){
      return response.json()
    }).then(function(body){
      // console.log(body);
      alert(curEmail);
    })
    .catch(err => console.log(err));
    
    this.username.value ="";
    this.email.value="";
    this.checkbox1.checked=false;
    this.checkbox2.checked=false;
    this.props.history.push('/ParticipantSignUpPending');

      }

  render() {
      let email = this.state.email;

      return (
        <div className="mainpage">

        {!email &&
        <div className="wrappermainpage">      

            <div className="wrapperpage2">
            <div style={{width: 'auto', margin: 'auto', color:"dimgray"}}>
            <Step.Group ordered size="mini" >
              <Step active >
                <Step.Content>
                  <Step.Title>Enrolling</Step.Title>
                  <Step.Description>Insert your email </Step.Description>
                </Step.Content>
              </Step>

              <Step disabled >
                <Step.Content>
                  <Step.Title>Confirmed Email </Step.Title>
                  <Step.Description>Welcome to our Survey</Step.Description>
                </Step.Content>
              </Step>

              <Step disabled >
                <Step.Content>
                  <Step.Title>Survey Questions</Step.Title>
                  <Step.Description>Please answer questions</Step.Description>
                </Step.Content>
              </Step>

              <Step disabled >
                <Step.Content>
                  <Step.Title>Rating and Further Experiences</Step.Title>
                  <Step.Description>Please rate our survey</Step.Description>
                </Step.Content>
              </Step>

              <Step disabled >
                <Step.Content>
                  <Step.Title>Survey Submitted</Step.Title>
                  <Step.Description>Thanks for your participation</Step.Description>
                </Step.Content>
              </Step>
            </Step.Group>
            </div>
         
            <Segment >
           
              <form  onSubmit={this.handleSubmit}> 
                <table>
                  <p className="ppage2" > Please insert your name and email address(Name just used as alias)
                  </p>
                  <div className="form-group-p2">
                  <Icon name='user' style={{color:"black"}}/>
                  <input 
                      placeholder= "Name"
                      type="username" 
                      name="username" 
                      className="form-control-2"
                      ref={(a) => this.username = a}
                      required/>
                  </div>
                  <div className="form-group-p2">
                  <Icon name='mail' style={{color:"black"}} />
                    <input 
                      placeholder= "Email"
                      type="email" 
                      name="email" 
                      className="form-control-2"
                      ref={(a) => this.email = a}
                      required/>
                  </div>
                 <div className="form-group-pag2">
                 <label className="labelpage1" style={{fontFamily:  "sans-serif", fontSize:'15px'}}>
                  <input 
                     type="checkbox" 
                     name="checkbox1" 
                     className="form-control-12"
                     ref={(a) => this.checkbox1 = a}
                     required />
                     I agree to the 
                     <Modal
                      trigger={<a style={{fontFamily:  "sans-serif", fontSize:'15px', color:"blue"}}> {' '}Terms & Use
                     </a>}
                      header='Pollinator Insect Online Survey'
                      content={<p style={{ padding: '15px', textAlign:'justify'}}>Thank you for interest in participating in this survey. This survey is run by 
                                      the University of New England ...
                                       <br/>
                                      In this survey you will be shown a series of images of insect pollinators, 
                                      and will be asked to identify the type of insect shown in each by selecting multiple choice answers.  </p>}
                      actions={[
                        
                          { key: 'done', content: 'Agree', positive: true },
                        ]}
                      style={{ position: 'absolute',
                              top: '50%',
                              left: '50%',
                              transform: 'translate(-50%, -50%)',
                              textAlign:'justify'}}
                      />
                     , including our Cookie use
                      </label>
                </div>
                <div className="form-group-pag2">
                  <input 
                     type="checkbox" 
                     name="checkbox2" 
                     className="form-control-12"
                     ref={(a) => this.checkbox2 = a}
                     required />
                     <label className="labelpage1" style={{fontFamily:  "sans-serif", fontSize:'15px'}}>
                      I have not undertaken</label>
                </div>
                <button 
                  type="submit" 
                  className="btn-primary21"
                  style={{fontFamily:  "sans-serif", fontSize:'15px', paddingLeft:"40px"}}
                  >
                  Send Email
                </button>
                </table>
                </form>
                </Segment>
              </div>
          </div>
    }
          </div>
      );
    }
  }

Page2.defaultProps = {
  history: {},
  userDelete: () => {},
  showDelete: () => {},
  // email: '',
  // name: '',
  handleSubmit: () => {},
  closeModal: () => {},
  openModal: () => {}
}
 
// prop checks
Page2.propTypes = {
  // modalIsOpen: PropTypes.bool.isRequired,
  history: PropTypes.object.isRequired,
  openModal: PropTypes.func.isRequired,
  userDelete: PropTypes.func.isRequired,
  // email: PropTypes.string.isRequired,
  // name: PropTypes.string.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  // afterOpenModal: PropTypes.func.isRequired,
  closeModal: PropTypes.func.isRequired
};

export default Page2;