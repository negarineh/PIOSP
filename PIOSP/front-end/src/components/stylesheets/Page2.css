.form-control-2 {
  width: 80%;
  font-size: 18px;
  font-weight: bold;
  text-align: center;
  border-radius: 5px;
  margin-bottom: 20px;
  height: 30px;
  font-family: 'Times New Roman', Times, serif;
  padding: 1px; }

.form-control-12 {
  border-radius: 5px;
  font-size: 18px;
  font-weight: bold;
  width: 20px;
  height: 20px; }

.form-group-pag2 {
  padding-left: 18px; }

.btn-primary21 {
  position: relative;
  display: block;
  margin-left: 20px;
  padding: 10px 30px;
  margin-top: 10px;
  font-size: 20px;
  font-family: "fantasy";
  font-weight: 400;
  color: white;
  background-image: linear-gradient(#38cf38, #21ca0a);
  border: none;
  z-index: 1;
  border-radius: 5px; }
  .btn-primary21:before {
    position: absolute;
    content: '';
    top: 0;
    left: 0;
    width: 0;
    height: 100%;
    background-image: linear-gradient(#0a830a, #21ca0a);
    transition: width 250ms;
    z-index: -1;
    border-radius: 10px; }
  .btn-primary21:hover {
    color: white; }
    .btn-primary21:hover:before {
      width: 100%; }

.labelpage1 {
  color: #463831;
  font-family: sans-serif;
  font-size: 15px; }

.ppage2 {
  display: flex;
  text-align: left;
  height: 70%;
  width: 100%;
  color: #463831;
  line-height: 1.5;
  font-family: sans-serif;
  font-size: 15px; }

.fieldsetpage2 {
  text-align: left;
  padding: 20px;
  padding-bottom: 20px;
  margin-top: 20px;
  margin-left: 20px;
  margin-right: 20px;
  margin-bottom: 20px;
  border-radius: 10px; }

.pmodalpage2 {
  text-align: justify;
  color: #463831;
  font-family: sans-serif;
  font-size: 15px;
  height: 100%;
  width: 100%; }

.legendpage2 {
  font-family: "fantasy";
  font-size: 20px;
  text-align: center; }

.h2page2 {
  text-align: center;
  color: darkslategray;
  margin-bottom: 20px; }

.fa fa-spinner fa-pulse fa-3x fa-fw {
  justify-content: center; }

.setheader {
  padding-left: 190px;
  padding-bottom: 10px; }

.wrapperpage2 {
  padding: 10px;
  margin: 10px; }
