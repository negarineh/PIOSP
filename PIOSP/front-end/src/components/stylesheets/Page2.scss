 // Variables
$color-bg: white;
$color-main: #333;
$color-hover: lighten(#0275d8, 1%);
$padding: 20px;
$radius: 10px;
$accent-color: #dfcfad; 
$font-size:20px;
$font-family: 'fantasy';
$background-color: rgb(78, 82, 76);
$background-color-after: #e74c3c;

// @font-face {
//   font-family: 'Font-Awesome';
//   src: url('../fonts/font-awesome-4.7.0/fonts/fontawesome-webfont.woff2') format("woff2");;
// }

// Mixins
@mixin transition($prop: all, $duration: 0.3s, $timing: ease-in-out) {
  transition: $prop $duration $timing;
}

// h1 {
//   color: $color-main;
//   font-size: 30px;
//   margin-top: 0;
//   text-align: center; 
//   font-family: 'fantasy';
// }
 
.form-control-2 {  
//   border: none;
//   border-bottom: 1px solid $color-main;
//   background: none;
  width: 80%;
  font-size: 18px;
  font-weight: bold;
//   line-height: 1.2;
//   outline: none;
  text-align: center;
  border-radius: 5px;
//   vertical-align: bottom;
//   position: relative;
//   // float: left;
//   // *overflow: hidden;
    margin-bottom: $padding;
    // size: 20px;
    height:30px;
    font-family: 'Times New Roman', Times, serif;
    padding: 1px;
} 

.form-control-12{
  border-radius: 5px;
  font-size: 18px;
  font-weight: bold;
  width: 20px;
  height:20px;
}

.form-group-pag2{
  // display: inline-block;
  padding-left: 18px;
}

// .btn:hover {
//   background: $color-hover;
//   border-color: $color-hover;
// }
   
// .wrapperpage1 {      
//   margin: 50px;
//   font-size: 20px;
//   // text-align: center;
//   width: 1200px; 
//   height: 550px;
//   border-radius: $radius;
//   background: transparent;
//   box-shadow: 0 15px 35px rgba(28, 28, 204, 0.1), 0 5px 15px rgba(0, 0, 0, 0.514);
//   // padding: $padding;
//   background-color: white;
//   }  

.btn-primary21{
			position: relative;
			display: block;
      // margin: 30px auto 0;
      margin-left: 20px;
      padding: 10px 30px;
      margin-top: 10px;
			font-size: $font-size;
			font-family: $font-family;
			font-weight: 400;
			color: white;
			background-image: linear-gradient(#38cf38,#21ca0a);
			border: none;
			z-index: 1;
			border-radius: 5px;
			&:before {
				position: absolute;
				content: '';
				top: 0;
				left: 0;
				width: 0;
				height: 100%;
				background-image: linear-gradient(#0a830a,#21ca0a);
				transition: width 250ms;
				z-index: -1;
				border-radius: $radius;
			}
			
			&:hover {
				color: white;
				
				&:before {
					width: 100%;
				}
			}
  }

.labelpage1{
  color: rgb(70, 56, 49);
  // background: $accent-color;
  font-family:  sans-serif ;
  font-size: 15px; 
//   display: block; 
}
 
.ppage2{
  display: flex;
//   align-items: center ;
  text-align: left;
//   justify-content: center;
  // position: absolute;
  height: 70%; 
  width: 100%;
  color: rgb(70, 56, 49);
  // padding-bottom: $padding;
  line-height: 1.5;
  font-family:  sans-serif ;
  font-size: 15px;
} 

.fieldsetpage2{
  text-align: left;
  padding: $padding; 
  padding-bottom: 20px;
  margin-top: $padding;  
  margin-left: $padding;
  margin-right: $padding;
  margin-bottom: $padding;
  border-radius: $radius; 
}

.pmodalpage2{
  text-align: justify;
  color: rgb(70, 56, 49);
  font-family:  sans-serif ;
  font-size: 15px;
  height: 100%; 
  width: 100%;
}

.legendpage2{
	// padding-top: 10px;
	font-family: $font-family;
    font-size: $font-size;
    text-align: center;
}

// .wrapper1.input{
//   align-self: auto;
//   float: left;
// }
// .wrapper1.label{
//   text-align: right;
//   float: left;
// }

// .form-group{
//   padding: 5px 5px 5px 5px;
//   left: 60px;
//   position: absolute;
// }

// .loader {
//   position: relative;
//   top: 50%;
//   right: 50%;
//   bottom: 50%;
//   left: 50%;
//   background: white;
//   z-index: 9999;
//   background-color: transparent;
//   color: grey;
// }

.h2page2{
  text-align: center;
  color: darkslategray;
  margin-bottom: 20px;
}

.fa fa-spinner fa-pulse fa-3x fa-fw{
  justify-content: center;
}

.setheader{
  // position: relative;
  padding-left: 190px;
  padding-bottom: 10px;
}

.wrapperpage2{
  padding: 10px;
  margin: 10px;
}