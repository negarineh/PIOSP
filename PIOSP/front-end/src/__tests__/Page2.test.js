import React from "react";
import Enzyme, { shallow } from "enzyme";
import assert from "assert";
import Page2 from "../components/Survey/Page2";
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
 
Enzyme.configure({ adapter: new Adapter() });

// unit tests for the UserDeletePrompt component
describe('UserDeletePrompt component', () => {
  describe('render()', () => {
    it('should render the component', () => {
      const props = {history: {}, userDelete: ()=>{}, shoeDelete: ()=>{}, handleSubmit: () => {}, closeModal: () => {}, openModal: () => {}};
      const wrapper = shallow(<Page2 {...props}/>);
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
      assert.equal(wrapper.length, 1);
    });
  });
});
