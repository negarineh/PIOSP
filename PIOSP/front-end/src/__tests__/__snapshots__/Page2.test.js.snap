// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDeletePrompt component render() should render the component 1`] = `
<div
  className="mainpage"
>
  <div
    className="wrappermainpage"
  >
    <div
      className="wrapperpage2"
    >
      <div
        style={
          Object {
            "color": "dimgray",
            "margin": "auto",
            "width": "auto",
          }
        }
      >
        <StepGroup
          ordered={true}
          size="mini"
        >
          <Step
            active={true}
          >
            <StepContent>
              <StepTitle>
                Enrolling
              </StepTitle>
              <StepDescription>
                Insert your email 
              </StepDescription>
            </StepContent>
          </Step>
          <Step
            disabled={true}
          >
            <StepContent>
              <StepTitle>
                Confirmed Email 
              </StepTitle>
              <StepDescription>
                Welcome to our Survey
              </StepDescription>
            </StepContent>
          </Step>
          <Step
            disabled={true}
          >
            <StepContent>
              <StepTitle>
                Survey Questions
              </StepTitle>
              <StepDescription>
                Please answer questions
              </StepDescription>
            </StepContent>
          </Step>
          <Step
            disabled={true}
          >
            <StepContent>
              <StepTitle>
                Rating and Further Experiences
              </StepTitle>
              <StepDescription>
                Please rate our survey
              </StepDescription>
            </StepContent>
          </Step>
          <Step
            disabled={true}
          >
            <StepContent>
              <StepTitle>
                Survey Submitted
              </StepTitle>
              <StepDescription>
                Thanks for your participation
              </StepDescription>
            </StepContent>
          </Step>
        </StepGroup>
      </div>
      <Segment>
        <form
          onSubmit={[Function]}
        >
          <table>
            <p
              className="ppage2"
            >
               Please insert your name and email address(Name just used as alias)
            </p>
            <div
              className="form-group-p2"
            >
              <Icon
                as="i"
                name="user"
                style={
                  Object {
                    "color": "black",
                  }
                }
              />
              <input
                className="form-control-2"
                name="username"
                placeholder="Name"
                required={true}
                type="username"
              />
            </div>
            <div
              className="form-group-p2"
            >
              <Icon
                as="i"
                name="mail"
                style={
                  Object {
                    "color": "black",
                  }
                }
              />
              <input
                className="form-control-2"
                name="email"
                placeholder="Email"
                required={true}
                type="email"
              />
            </div>
            <div
              className="form-group-pag2"
            >
              <label
                className="labelpage1"
                style={
                  Object {
                    "fontFamily": "sans-serif",
                    "fontSize": "15px",
                  }
                }
              >
                <input
                  className="form-control-12"
                  name="checkbox1"
                  required={true}
                  type="checkbox"
                />
                I agree to the
                <Modal
                  actions={
                    Array [
                      Object {
                        "content": "Agree",
                        "key": "done",
                        "positive": true,
                      },
                    ]
                  }
                  closeOnDimmerClick={true}
                  closeOnDocumentClick={false}
                  content={
                    <p
                      style={
                        Object {
                          "padding": "15px",
                          "textAlign": "justify",
                        }
                      }
                    >
                      Thank you for interest in participating in this survey. This survey is run by the University of New England ...
                      <br />
                      In this survey you will be shown a series of images of insect pollinators, and will be asked to identify the type of insect shown in each by selecting multiple choice answers.  
                    </p>
                  }
                  dimmer={true}
                  eventPool="Modal"
                  header="Pollinator Insect Online Survey"
                  style={
                    Object {
                      "left": "50%",
                      "position": "absolute",
                      "textAlign": "justify",
                      "top": "50%",
                      "transform": "translate(-50%, -50%)",
                    }
                  }
                  trigger={
                    <a
                      style={
                        Object {
                          "color": "blue",
                          "fontFamily": "sans-serif",
                          "fontSize": "15px",
                        }
                      }
                    >
                       
                       
                      Terms & Use
                    </a>
                  }
                />
                , including our Cookie use
              </label>
            </div>
            <div
              className="form-group-pag2"
            >
              <input
                className="form-control-12"
                name="checkbox2"
                required={true}
                type="checkbox"
              />
              <label
                className="labelpage1"
                style={
                  Object {
                    "fontFamily": "sans-serif",
                    "fontSize": "15px",
                  }
                }
              >
                I have not undertaken
              </label>
            </div>
            <button
              className="btn-primary21"
              style={
                Object {
                  "fontFamily": "sans-serif",
                  "fontSize": "15px",
                  "paddingLeft": "40px",
                }
              }
              type="submit"
            >
              Send Email
            </button>
          </table>
        </form>
      </Segment>
    </div>
  </div>
</div>
`;
